code/game/machinery/suit_storage_unit.dm

```diff
@@ -1,184 +1,406 @@
 
 /obj/machinery/suit_storage_unit/cmo
-	SUIT_TYPE = /obj/item/clothing/suit/space/hardsuit/medical
-	HELMET_TYPE = /obj/item/clothing/head/helmet/space/hardsuit/medical
-	MASK_TYPE = /obj/item/clothing/mask/breath
-
-/obj/machinery/suit_storage_unit/New()
-	src.update_icon()
-	if(SUIT_TYPE)
-		SUIT = new SUIT_TYPE(src)
-	if(HELMET_TYPE)
-		HELMET = new HELMET_TYPE(src)
-	if(MASK_TYPE)
-		MASK = new MASK_TYPE(src)
+	suit_type    = /obj/item/clothing/suit/space/hardsuit/medical
+	helmet_type  = /obj/item/clothing/head/helmet/space/hardsuit/medical
+	mask_type    = /obj/item/clothing/mask/breath
+
+/obj/machinery/suit_storage_unit/syndicate
+	suit_type    = /obj/item/clothing/suit/space/hardsuit/syndi
+	helmet_type  = /obj/item/clothing/head/helmet/space/hardsuit/syndi
+	mask_type    = /obj/item/clothing/mask/gas/syndicate
+	storage_type = /obj/item/tank/jetpack/oxygen/harness
+
+/obj/machinery/suit_storage_unit/ert/command
+	suit_type    = /obj/item/clothing/suit/space/hardsuit/ert/commander
+	helmet_type  = /obj/item/clothing/head/helmet/space/hardsuit/ert/commander
+	mask_type    = /obj/item/clothing/mask/breath
+	storage_type = /obj/item/tank/emergency_oxygen/double
+
+/obj/machinery/suit_storage_unit/ert/security
+	suit_type    = /obj/item/clothing/suit/space/hardsuit/ert/security
+	helmet_type  = /obj/item/clothing/head/helmet/space/hardsuit/ert/security
+	mask_type    = /obj/item/clothing/mask/breath
+	storage_type = /obj/item/tank/emergency_oxygen/double
+
+/obj/machinery/suit_storage_unit/ert/engineer
+	suit_type    = /obj/item/clothing/suit/space/hardsuit/ert/engineer
+	helmet_type  = /obj/item/clothing/head/helmet/space/hardsuit/ert/engineer
+	mask_type    = /obj/item/clothing/mask/breath
+	storage_type = /obj/item/tank/emergency_oxygen/double
+
+/obj/machinery/suit_storage_unit/ert/medical
+	suit_type    = /obj/item/clothing/suit/space/hardsuit/ert/medical
+	helmet_type  = /obj/item/clothing/head/helmet/space/hardsuit/ert/medical
+	mask_type    = /obj/item/clothing/mask/breath
+	storage_type = /obj/item/tank/emergency_oxygen/double
+
+
+
+/obj/machinery/suit_storage_unit/Initialize()
+	. = ..()
+	if(suit_type)
+		suit = new suit_type(src)
+	if(helmet_type)
+		helmet = new helmet_type(src)
+	if(mask_type)
+		mask = new mask_type(src)
+	if(storage_type)
+		storage = new storage_type(src)
+	update_icon()
+
+	//move this into machinery eventually...
+	if(occupant_typecache)
+		occupant_typecache = typecacheof(occupant_typecache)
+
+/obj/machinery/suit_storage_unit/Destroy()
+	QDEL_NULL(suit)
+	QDEL_NULL(helmet)
+	QDEL_NULL(mask)
+	QDEL_NULL(storage)
+	return ..()
 
 /obj/machinery/suit_storage_unit/update_icon()
-	var/hashelmet = 0
-	var/hassuit = 0
-	var/hashuman = 0
-	if(HELMET)
-		hashelmet = 1
-	if(SUIT)
-		hassuit = 1
-	if(OCCUPANT)
-		hashuman = 1
-	icon_state = text("suitstorage[][][][][][][][][]",hashelmet,hassuit,hashuman,src.isopen,src.islocked,src.isUV,src.ispowered,src.isbroken,src.issuperUV)

```

Comment:
:trollface:

Reply:
the worst part is there are more instances of this shittery in the codebase

Reply:
the worst part is there are more instances of this shittery in the codebase